lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -40,7 +40,7 @@ def interpret_json_data(pod_data)
       elsif @phase == "Terminating"
         @status = @phase
       else
-        ready_condition = pod_data["status"]["conditions"].find { |condition| condition["type"] == "Ready" }
+        ready_condition = pod_data["status"].fetch("conditions", {}).find { |condition| condition["type"] == "Ready" }
```

Comment:
I was a little confused at first for the `#find` but I can see in [the docs](https://kubernetes.io/docs/api-reference/v1.5/#deploymentcondition-v1beta1) that `conditions` is an array. I assume it's an array of all possible conditions, i.e. `type=ready`. What are some of the others? In this case you indeed care about that one.

It's definitely puzzling why `status` wouldn't always have conditions.

However, since `conditions` is an array shouldn't this be `#fetch("conditions", [])`? Otherwise the `#find` might behave strangely.